connectors/spark/src/main/scala/io/unitycatalog/spark/UCSingleCatalog.scala

```diff
@@ -39,11 +39,20 @@ class UCSingleCatalog
     if (urlStr == null) {
       throw new IllegalArgumentException(s"uri must be specified for Unity Catalog '$name'")
     }
+
     val url = new URI(urlStr)
-    apiClient = new ApiClient()
-      .setHost(url.getHost)
-      .setPort(url.getPort)
-      .setScheme(url.getScheme)
+    apiClient = {
```

Comment:
alternatively, couldn't this whole change just really be 
```
    Option(url.getPath).filter(_.nonEmpty).foreach(apiClient.setBasePath)
```
idk if that's more readable.